### PR TITLE
Shop Interface

### PR DESCRIPTION
**Pull request by hannahfolkertsma**

### CODE REVIEW

File: Project/Fall2020_CSC403_Project/TextEngine.cs [hunk 1]
```diff
@@ -224,6 +271,35 @@ private void HandleMarkup(string line)
                 case Markup.CheckThresholdsForTree:
                     // line is: empty, just check thresholds
                     break;
+                case Markup.ExitOptions:
+                    //  HandleMarkup(Story.GetNextLine());
```
Comment (HayleyO) [inline Project/Fall2020_CSC403_Project/TextEngine.cs:275]: It'd be good if you added a comment here saying what the line being read in looked like and what would happen

File: Project/Fall2020_CSC403_Project/TextEngine.cs [hunk 2]
```diff
@@ -224,6 +271,35 @@ private void HandleMarkup(string line)
...
+                    isShop = false;
+                    break;
+                case Markup.HannahShopOptions:
+                    string shop1OptionString = "#O S ";
```
Comment (HayleyO) [inline Project/Fall2020_CSC403_Project/TextEngine.cs:279]: Comment here with what this line looks like exactly and what's happening

File: Project/Fall2020_CSC403_Project/TextEngine.cs [hunk 3]
```diff
@@ -224,6 +271,35 @@ private void HandleMarkup(string line)
...
+                        }
+                    }
+
+                    shop1OptionString += " Exit,#E]";
+                    Console.WriteLine(shop1OptionString);
+                    Story.CurrentStoryText.AddFirst(shop1OptionString);
+                    HandleMarkup(Story.GetNextLine());
+                    break;
+                case Markup.HayleyShopOptions:
+                    string shop2OptionString = "#O S ";
```
Comment (HayleyO) [inline Project/Fall2020_CSC403_Project/TextEngine.cs:294]: This comment is another comment about comments

File: Project/Fall2020_CSC403_Project/TextEngine.cs [hunk 4]
```diff
@@ -188,16 +211,40 @@ private void HandleMarkup(string line)
                     }
                     this.Options.Peek().OptionFocused = true; //Focus the top option 
                     this.ChangeText(newLineOptions);
+                    options.Reverse();
```
Comment (HayleyO) [inline Project/Fall2020_CSC403_Project/TextEngine.cs:214]: I am still not sure about this

Comment (HayleyO) [reply Project/Fall2020_CSC403_Project/TextEngine.cs:214]: I think there must be some kind of ambiguity with options naming happening, but if it works it works and I won't harp on it

File: Project/Fall2020_CSC403_Project/TextEngine.cs [hunk 5]
```diff
@@ -1,37 +1,37 @@
...
-        private int originalWidth { get; set; }
+        private Stack<Option> TempOptions = new Stack<Option>();
+        private List<Option> options = new List<Option>();
+        public double ForegroundImage_Xscale { get; set; }
+        public double ForegroundImage_Yscale { get; set; }
+        public double ForegroundImage_AspectRatio { get; set; }
+        public int originalHeight { get; set; }
+        public int originalWidth { get; set; }
+        private bool isShop { get; set; } = false; 
+        private string line { get; set; }
```
Comment (HayleyO) [inline Project/Fall2020_CSC403_Project/TextEngine.cs:24]: I'm not sure why these were moved around but it seems outside the scope of this issue

**Review by HayleyO — APPROVED** [review]
Overall very minor things. Looks like it should work fine, though, although keep my comments in mind.

File: Project/Fall2020_CSC403_Project/TextEngine.cs (hunk 4 above)
Comment (HayleyO) [reply Project/Fall2020_CSC403_Project/TextEngine.cs:214]: I think since it's in a switch case there's ambiguity in this case from previous case with something named options

Comment (HayleyO) [reply Project/Fall2020_CSC403_Project/TextEngine.cs:214]: That could be a problem in the future to look at

File: Project/Fall2020_CSC403_Project/TextEngine.cs (hunk 5 above)
Comment (hannahfolkertsma) [reply Project/Fall2020_CSC403_Project/TextEngine.cs:24]: moved these around to address some UI issues i was having when i made the options 2 columns!

File: Project/Fall2020_CSC403_Project/TextEngine.cs (hunk 1 above)
Comment (hannahfolkertsma) [reply Project/Fall2020_CSC403_Project/TextEngine.cs:275]: gonna add comments :)

File: Project/Fall2020_CSC403_Project/TextEngine.cs (hunk 4 above)
Comment (hannahfolkertsma) [reply Project/Fall2020_CSC403_Project/TextEngine.cs:214]: It is kind of weird but i dont think there are any negative effects to reversing the options list back to the way it was since it has already done its job of filling the stack at this point
